obspy/core/tests/test_waveform_plugins.py

```diff
@@ -53,13 +53,16 @@ def test_read_and_write(self):
         Tests read and write methods for all waveform plug-ins.
         """
         data = np.arange(0, 2000)
-        start = UTCDateTime(2009, 1, 13, 12, 1, 2, 999000)
         formats = _get_default_eps('obspy.plugin.waveform', 'writeFormat')
         for format in formats:
+            start = UTCDateTime(2009, 1, 13, 12, 1, 2, 999000)
             # XXX: skip SEGY and SU formats for now as they need some special
             # headers.
             if format in ['SEGY', 'SU', 'SEG2']:
                 continue
+            elif format in ['GCF']:
+               # XXX: GCF format does not support fractional start time for sampling rates <= 250 Hz
+               start = UTCDateTime(2009, 1, 13, 12, 1, 3)
```

Comment:
Any particular reason why GCF can not be tested with split seconds? We use a wonky number for a reason :)

Reply:
That's a format restriction, we probably have to ask Guralp why this is so, but if I'm to elaborate a bit: The gcf data files consists of 1024 byte blocks, each having a header of 16-byte header leaving 1008 byte for the data. The lowest compression (using signed 32-bit first difference) can then at maximum store 250 data points. So basically for any sampling rate < 250 Hz you can fit an integer number of whole seconds of data into each block. To keep it neat and clean I guess at the time designing the format specification they simply decided that all time-series should sample at integer seconds and for sampling rates < 250 Hz each block should hold whole seconds data segments. Note that even for sampling rates > 250 Hz the start-time is restricted. e.g. 500 Hz data may only start at whole or half seconds (i.e. 0.00s or 0.50s)

Reply:
:+1: